daemon/control.c

```diff
@@ -672,24 +633,40 @@ control_handle_container_start(control_t *control, container_t *container,
 }
 
 static bool
-control_check_command(const ControllerToDaemon *msg)
+control_check_command(control_t *control, const ControllerToDaemon *msg)
 {
+	// Device is not yet provisioned or in hosted mode, alllow all commands
 	if (!cmld_is_device_provisioned() || cmld_is_hostedmode_active()) {
-		DEBUG("Device is not provisioned or in hosted mode, all commands are accepted");
+		TRACE("Device is not provisioned or in hosted mode, all commands are accepted");
 		return true;
 	}
 
-	if ((msg->command == CONTROLLER_TO_DAEMON__COMMAND__LIST_CONTAINERS) ||
-	    (msg->command == CONTROLLER_TO_DAEMON__COMMAND__CONTAINER_CHANGE_TOKEN_PIN) ||
-	    (msg->command == CONTROLLER_TO_DAEMON__COMMAND__CREATE_CONTAINER) ||
-	    (msg->command == CONTROLLER_TO_DAEMON__COMMAND__CONTAINER_START) ||
-	    (msg->command == CONTROLLER_TO_DAEMON__COMMAND__CONTAINER_UPDATE_CONFIG) ||
-	    (msg->command == CONTROLLER_TO_DAEMON__COMMAND__GET_CONTAINER_STATUS) ||
-	    (msg->command == CONTROLLER_TO_DAEMON__COMMAND__CONTAINER_CMLD_HANDLES_PIN) ||
-	    (msg->command == CONTROLLER_TO_DAEMON__COMMAND__CONTAINER_STOP)) {
-		TRACE("Received command %d is valid in provisioned mode", msg->command);
-		return true;
+	if (control->privileged) {
+		// Device is in privileged provisioned mode, only allow subset of commands
+		if ((msg->command == CONTROLLER_TO_DAEMON__COMMAND__LIST_CONTAINERS) ||
+		    (msg->command == CONTROLLER_TO_DAEMON__COMMAND__CONTAINER_CHANGE_TOKEN_PIN) ||
+		    (msg->command == CONTROLLER_TO_DAEMON__COMMAND__CREATE_CONTAINER) ||
+		    (msg->command == CONTROLLER_TO_DAEMON__COMMAND__CONTAINER_START) ||
+		    (msg->command == CONTROLLER_TO_DAEMON__COMMAND__CONTAINER_UPDATE_CONFIG) ||
+		    (msg->command == CONTROLLER_TO_DAEMON__COMMAND__GET_CONTAINER_STATUS) ||
+		    (msg->command == CONTROLLER_TO_DAEMON__COMMAND__CONTAINER_CMLD_HANDLES_PIN) ||
+		    (msg->command == CONTROLLER_TO_DAEMON__COMMAND__CONTAINER_STOP)) {
+			TRACE("Received command %d is valid in provisioned mode", msg->command);
+			return true;
+		}
+	} else {
+		// Device is in unprivileged provisioned mode, only allow subset of commands
```

Comment:
logic is wrong! Check priviliged flag first so taht only these three commands are allowed,
for unpriv control interface regardless if hosted mode or provisioned state

Reply:
Done